### PR TITLE
[GPU][DG2] Fix some unit tests

### PR DESCRIPTION
**Pull request by tuxedcat**
### Related Test Cases:
 - BinaryConvTest/binary_convolution_test.conv/0
 - binary_convolution.basic_convolution_1x1_single_packed_channel
   - Refer #12486
 - fusings_gpu/conv_scale_activation_eltwise_fp32_quantize_i8.basic/4
 - fusings_gpu/pooling_scale_activation_quantize.per_channel/0
 - fusings_gpu/mvn_scale_activation_eltwise_fp32_quantize_i8.basic/0
 - fusings_gpu/mvn_activation.basic/0
 - fusings_gpu/conv_scale_activation_eltwise_fp32_quantize_i8.basic/0
   - Missed case from #14325
 - fusings_gpu/deconv_actv_eltw_actv.basic/20
 - fusings_gpu/conv_int8_activation_eltwise.fsv16/0
 - fusings_gpu/conv_int8_activation_eltwise_quantize.fsv16/0
 - fusings_gpu/conv_fp32_wrong_bias.basic/0
   - Remove WA(#13978) and use p.expected_fused_primitives_onednn
 - fusings_gpu/deconv_scale.fp16_scale_out/19
   - Limit input value smaller to avoid floating point error
 - fusings_gpu/conv_int8_prelu_eltwise.basic_slope_2/20
   - Increase tolerance for int8 topology
 - fusings_gpu/conv_fp32_prelu_eltwise.basic_sum/3
   - Change tolerance from default_type to data_type
 - fusings_gpu/conv_int8_eltwise_onednn.u8_eltwise_sum_out/0
   - WA: Remove meaningless crop to avoid accuracy error. Needs follow-up analyze.

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 1]
```diff
@@ -1799,7 +1799,7 @@ TEST_P(conv_int8_prelu_eltwise, basic) {
         reorder("reorder_bfyx", input_info("eltwise"), p.default_format, data_types::f32)
     );
 
-    tolerance = default_tolerance(p.default_type);
+    tolerance = 1;
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:1802]:
```
    tolerance = default_tolerance(p.data_type);
```

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 2]
```diff
@@ -1817,7 +1817,7 @@ TEST_P(conv_int8_prelu_eltwise, basic_slope_2) {
         reorder("reorder_bfyx", input_info("eltwise"), p.default_format, data_types::f32)
     );
 
-    tolerance = default_tolerance(p.default_type);
+    tolerance = 1.5;
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:1820]: Let's change implementation in `default_tolerance`

File: src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp [hunk 3]
```diff
@@ -596,81 +564,81 @@ TEST_P(deconv_scale_actv_quant_i8, basic) {
...
-    deconv_test_params{ CASE_DECONV_FP16_4, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_FP16_5, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_FP16_6, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_FP16_7, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_FP16_8, 2, 2, 5 },
-
-    deconv_test_params{ CASE_DECONV_U8S8_1, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_U8S8_2, 2, 2, 5 },
-    deconv_test_params{ CASE_DECONV_U8S8_3, 2, 2, 5 },
+    deconv_test_params{ CASE_DECONV_FP32_1, 2, 3, 5 },
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp:567]: Please leave an explanation in commit message.

**Review by isanghao — CHANGES_REQUESTED** [review]
Looks good in general, but some comments are left.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 1 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 2 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp (hunk 3 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp:567]: Okay. I'll try to write commit message more detailed.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 4]
```diff
@@ -3066,37 +3056,36 @@ TEST_P(conv_int8_eltwise_onednn, u8_eltwise_prod_out) {
...
-
-    convolution_test_params{ CASE_CONV3D_U8S8_1, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_U8S8_2, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_U8S8_3, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_U8S8_5, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_S8S8_1, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_S8S8_2, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_S8S8_3, 3, 4 },
-    convolution_test_params{ CASE_CONV3D_S8S8_5, 3, 4 },
+    convolution_test_params{ CASE_CONV_U8S8_1, 0, 2, 3 },
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3067]: #define NOT_FOR_CLDNN

File: src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp [hunk 5]
```diff
@@ -398,30 +379,30 @@ TEST_P(deconv_scale, basic) {
...
+        data("weights", get_mem(get_weights_layout(p), -4, 4)),
         data("scale_data", get_mem(get_per_channel_layout(p), 1.0f/p.kernel.count())),
         deconvolution("deconv", input_info("input"), { "weights" }, p.groups, p.stride, p.pad),
         eltwise("scale", { input_info("deconv"), input_info("scale_data") }, eltwise_mode::prod),
         reorder("out", input_info("scale"), p.default_format, data_types::f32)
     );
 
-    tolerance = 1e-5f;
-    execute(p);
+    tolerance = default_tolerance(data_types::f16);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp:389]: can we use `p.data_type` instead of `f16`?

File: src/plugins/intel_gpu/tests/test_utils/test_utils.cpp [hunk 6]
```diff
@@ -388,12 +388,12 @@ std::vector<cldnn::data_types> generic_test::test_data_types() {
 double default_tolerance(data_types dt) {
     switch (dt) {
     case data_types::f16:
-        return 1e-3;
+        return 1e-2;
     case data_types::f32:
-        return 1e-5;
+        return 1e-4;
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_utils/test_utils.cpp:393]: For fp16 and fp32, let's not change it unless it is really necessary

File: src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp [hunk 7]
```diff
@@ -180,6 +180,11 @@ class binary_convolution_test : public ::testing::TestWithParam<TestParams> {
 
 TEST_P(binary_convolution_test, conv) {
     auto& engine = get_test_engine();
+
+    //FIXME: https://github.com/openvinotoolkit/openvino/pull/12486
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp:184]: `// Onednn was not validated for binary convolution: https://github.com/openvinotoolkit/openvino/pull/12486`

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 4 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp (hunk 5 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/deconvolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/test_utils/test_utils.cpp (hunk 6 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/test_utils/test_utils.cpp]: Reverted related changes.

File: src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp (hunk 7 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 8]
```diff
@@ -1260,15 +1264,15 @@ TEST_P(conv_fp32_swish, basic) {
 }
 
 INSTANTIATE_TEST_SUITE_P(fusings_gpu, conv_fp32_swish, ::testing::ValuesIn(std::vector<convolution_test_params>{
-    // convolution_test_params{ CASE_CONV_FP32_1, 2, 4 },
-    convolution_test_params{ CASE_CONV_FP32_2, 2, 4 },
-    convolution_test_params{ CASE_CONV_FP32_3, 2, 4 },
-    convolution_test_params{ CASE_CONV_FP32_4, 2, 4 },
+    // convolution_test_params{ CASE_CONV_FP32_1, CASE_CONV_FP32_1, 2, 4
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:1267]: why `CASE_CONV_FP32_1` is repeated? (in other places, too)

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 9]
```diff
@@ -1297,20 +1301,20 @@ TEST_P(conv_fp32_eltwise_b_fs_zyx_fsv16, splitted_vector_ops) {
...
-    convolution_test_params{ CASE_CONV_FP16_9, 2, 3 },
-    convolution_test_params{ CASE_CONV_FP16_11, 2, 3 },
-    convolution_test_params{ CASE_CONV_FP16_12, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_6, 2, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_7, 2, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_8, 2, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_9, 2, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_11, 2, 2, 3 },
+    convolution_test_params{ CASE_CONV_FP32_12, 2, 2, 3 },
+    // convolution_test_params{ CASE_CONV_FP32_13, CASE_CONV_FP32_13, 2, 3
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:1310]: why did you remove the comment?

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 10]
```diff
@@ -3066,37 +3056,37 @@ TEST_P(conv_int8_eltwise_onednn, u8_eltwise_prod_out) {
...
-        crop("crop", input_info("scale"), get_output_layout(p).get_tensor(), { 0, 0, 0, 0 }),
-        reorder("reorder_bfyx", input_info("crop"), p.default_format, data_types::f32)
+        reorder("reorder_bfyx", input_info("scale"), p.default_format, data_types::f32)
     );
 
     tolerance = 1.f;
     execute(p);
 }
 
+static const int NOT_FOR_CLDNN=0;
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:3066]: nit: `NOT_FOR_CLDNN = 0` (space before and after parenthesis)

File: src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp [hunk 11]
```diff
@@ -420,6 +428,9 @@ TEST(binary_convolution, basic_convolution_1x1_single_packed_channel) {
 
 TEST(binary_convolution, basic_convolution_1x1_single_packed_channel_fp16) {
     auto& engine = get_test_engine();
+    // DG2 has an issue for binary convolution: https://github.com/openvinotoolkit/openvino/pull/12486
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp:431]: Let's use soft expression: `not validated for binary convolution`. Please don't mention product name directly here.

**Review by isanghao — COMMENTED** [review]
Looks good in general. minor comments are left.

File: src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp (hunk 11 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/test_cases/binary_convolution_gpu_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 10 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 9 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: Done.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp (hunk 8 above)
Comment (tuxedcat) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp]: It was mistake. Done.

**Review by isanghao — APPROVED** [review]
